helix-term/src/commands/typed.rs

```diff
@@ -2408,6 +2409,78 @@ fn redraw(
     Ok(())
 }
 
+fn move_buffer(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    ensure!(args.len() == 1, format!(":move takes one argument"));
+
+    let new_path_as_url = helix_lsp::Url::from_file_path(helix_core::path::get_canonicalized_path(
+        &PathBuf::from(args.first().unwrap().to_string()),
+    ))
+    .unwrap();
+
+    let doc = doc!(cx.editor);
+    let old_path_as_url = doc
+        .url()
+        .ok_or_else(|| anyhow!("Scratch buffer cannot be moved. Use :write instead"))?;
+
+    let edits: Vec<(
+        helix_lsp::Result<helix_lsp::lsp::WorkspaceEdit>,
+        OffsetEncoding,
+        String,
+    )> = doc
+        .language_servers()
+        .map(|lsp| {
+            (
+                lsp.prepare_file_rename(&old_path_as_url, &new_path_as_url),
+                lsp.offset_encoding(),
+                lsp.name().to_owned(),
+            )
+        })
+        .filter(|(f, _, _)| f.is_some())
+        .map(|(f, encoding, name)| (helix_lsp::block_on(f.unwrap()), encoding, name))
+        .collect();
+
+    for (lsp_reply, encoding, name) in edits {
+        match lsp_reply {
+            Ok(edit) => {
+                if let Err(e) = apply_workspace_edit(cx.editor, encoding, &edit) {
+                    log::error!(
+                        ":move command failed to apply edits from lsp {}: {:?}",
+                        name,
+                        e
+                    );
+                };
+            }
+            Err(e) => {
+                log::error!("LSP {} failed to treat willRename request: {:?}", name, e);
+            }
+        };
+    }
+
+    let doc = doc_mut!(cx.editor);
+    doc.set_path(Some(new_path_as_url.to_file_path().unwrap().as_path()));
+    if let Err(e) = std::fs::rename(
+        &old_path_as_url.to_file_path().unwrap(),
+        &new_path_as_url.to_file_path().unwrap(),
```

Comment:
a bit of a code cleanliness here: You convert path -> URL -> path with a n unwrap.

Its mostly fine but a bit inefficient/nonideomatic (avoid unwraps whenever so I don't need to think about whether this could panic or not).

Just store the paths in separate variables.

Reply:
Done !